core/state/database.go

```diff
@@ -193,7 +193,7 @@ func (fdb *ForkingDB) CopyTrie(t Trie) Trie {
 
 	if fdb.started {
 		overlay := fdb.VerkleDB.CopyTrie(t)
-		return trie.NewTransitionTree(mpt.(*trie.SecureTrie), overlay.(*trie.VerkleTrie))
+		return trie.NewTransitionTree(mpt.(*trie.SecureTrie), overlay.(*trie.VerkleTrie), false /* XXX */, "copied")
```

Comment:
the `false` is incorrect, there should be an accessor for whether or not the tree is a storage tree, see implementation below.